dd-java-agent/instrumentation/avro/src/main/java/datadog/trace/instrumentation/avro/SchemaExtractor.java

```diff
@@ -0,0 +1,171 @@
+package datadog.trace.instrumentation.avro;
+
+import datadog.trace.api.DDTags;
+import datadog.trace.bootstrap.instrumentation.api.AgentDataStreamsMonitoring;
+import datadog.trace.bootstrap.instrumentation.api.AgentSpan;
+import datadog.trace.bootstrap.instrumentation.api.AgentTracer;
+import datadog.trace.bootstrap.instrumentation.api.Schema;
+import datadog.trace.bootstrap.instrumentation.api.SchemaBuilder;
+import datadog.trace.bootstrap.instrumentation.api.SchemaIterator;
+import java.util.List;
+import org.apache.avro.Schema.Field;
+
+public class SchemaExtractor implements SchemaIterator {
+  public static final String serialization = "serialization";
+  public static final String deserialization = "deserialization";
```

Comment:
prefer upper-case for constant names:
```suggestion
  public static final String SERIALIZATION = "serialization";
  public static final String DESERIALIZATION = "deserialization";
```